src/librbd/Operations.cc

```diff
@@ -779,7 +779,9 @@ int Operations<I>::snap_rollback(const cls::rbd::SnapshotNamespace& snap_namespa
   }
 
   r = prepare_image_update();
-  if (r < 0) {
+  if (r == -EAGAIN) {
```

Comment:
@dillaman  I changed title to 'librbd'.
The users may not know or care about this waiting time related to the 'osd_client_watch_timeout' in osd conf. Another situation is that if the exclusive lock is really occupied by any other alive owner all the time, then although still rare events, do/while loop here would tend to be a dead loop. So I think the best way is to return to advise the users to try again, let the users to decide what to do next.